web3/_utils/module_testing/go_ethereum_admin_module.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+
+from web3.datastructures import (
+    AttributeDict,
+)
+
+
+class GoEthereumAdminModuleTest:
+    def test_add_peer(self, web3):
+        result = web3.geth.admin.add_peer(
+                'enode://f1a6b0bdbf014355587c3018454d070ac57801f05d3b39fe85da574f002a32e929f683d72aa5a8318382e4d3c7a05c9b91687b0d997a39619fb8a6e7ad88e512@1.1.1.1:30303',  # noqa: E501
+                )
+        assert result is True
+
+    def test_admin_datadir(self, web3, datadir):
+        result = web3.geth.admin.datadir()
+        assert result == datadir
+
+    def test_admin_node_info(self, web3):
+        result = web3.geth.admin.node_info()
+        expected = AttributeDict({
+            'id': '',
+            'name': '',
+            'enode': '',
+            'ip': '',
+            'ports': AttributeDict({}),
+            'listenAddr': '',
+            'protocols': AttributeDict({})
+        })
+        # Test that result gives at least the keys that are listed in `expected`
+        assert len(expected.keys() - result.keys()) == 0
```

Comment:
This feels pretty janky... let me know if there is a better way. Different Geth versions return different `AttributeDict`s

Reply:
integration tests are janky so seems fine to me.  You can do something like `assert not set(expected.keys())difference(result.keys)` but that's just a slightly different way to assert that they have the same keys.